gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +164,233 @@ def _get_prior_measurements_in_world_frame(
+        w_i2Ui1_dict: RelativeDirectionsDict,
+        w_i2Ui1_dict_tracks: RelativeDirectionsDict,
+        wRi_list: List[Optional[Rot3]],
+        i2Ti1_priors: Dict[Tuple[int, int], PosePrior],
+        absolute_pose_priors: List[Optional[PosePrior]],
+        scale_factor: float,
+    ):
+        """Runs the averaging optimization.
+
+        Args:
+            num_images: number of images.
+            w_i2Ui1_dict: Unit directions from i2 to i1 in world frame indexed by (i1, i2).
+            w_i2Ui1_dict_tracks: Directions from camera to track in world frame indexed by (track_id, camera_id).
+            wRi_list: camera rotations in world frame.
+            i2Ti1_priors: relative pose priors.
+            absolute_pose_priors: absolute pose priors.
+            scale_factor: scale factor for the esimated translations.
+
+        Returns:
+            List of camera translations in world frame, with as many entries as the number of images.
+        """
+        logger.info(
+            "Using {} track measurements and {} camera measurements".format(len(w_i2Ui1_dict_tracks), len(w_i2Ui1_dict))
+        )
+
+        noise_model = gtsam.noiseModel.Isotropic.Sigma(NOISE_MODEL_DIMENSION, NOISE_MODEL_SIGMA)
+        if self._robust_measurement_noise:
+            huber_loss = gtsam.noiseModel.mEstimator.Huber.Create(HUBER_LOSS_K)
+            noise_model = gtsam.noiseModel.Robust.Create(huber_loss, noise_model)
+
+        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_i2Ui1_dict_tracks, noise_model)
+
+        # Run the optimizer.
+        try:
+            algorithm = TranslationRecovery()
+            if len(i2Ti1_priors) > 0:
+                # scale is ignored here.
+                w_i1ti2_priors = self._binary_measurements_from_priors(i2Ti1_priors, wRi_list)
+                wti_initial = self.__get_initial_values(absolute_pose_priors)
+                wti_values = algorithm.run(w_i1Ui2_measurements, 0.0, w_i1ti2_priors, wti_initial)
+            else:
+                wti_values = algorithm.run(w_i1Ui2_measurements, scale_factor)
+        except TypeError as e:
+            # TODO(akshay-krishnan): remove once latest gtsam pip wheels updated.
+            logger.error("TypeError: {}".format(str(e)))
+            algorithm = TranslationRecovery(w_i1Ui2_measurements)
+            wti_values = algorithm.run(scale_factor)
+
+        # transforming the result to the list of Point3
+        wti_list: List[Optional[Point3]] = [None] * num_images
+        for i in range(num_images):
+            if wRi_list[i] is not None and wti_values.exists(C(i)):
+                wti_list[i] = wti_values.atPoint3(C(i))
+        return wti_list
+
     # TODO(ayushbaid): Change wTi_initial to Pose3.
     def run_translation_averaging(
         self,
         num_images: int,
         i2Ui1_dict: Dict[Tuple[int, int], Optional[Unit3]],
         wRi_list: List[Optional[Rot3]],
+        tracks_2d: List[SfmTrack2d] = None,
```

Comment:
should this type be `Optional[List[SfmTrack2d]]`? Can we run mypy to see if we missed any?